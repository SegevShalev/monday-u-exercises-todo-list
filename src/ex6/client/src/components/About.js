import React, { useState, useEffect } from "react";
import styles from "../style/About.module.css";

const MIN_ID = 1;
const MAX_ID = 898;

export default function About() {
  const [pokemonId, setPokemonId] = useState();

  const changePokemon = () => {
    let rand = parseInt(MIN_ID + Math.random() * (MAX_ID - MIN_ID));
    setPokemonId(rand);
  };

  useEffect(() => {
    changePokemon();
  }, []);

  setInterval(changePokemon, 2000);
  var min = 1;
  var max = 900;
  var rand = parseInt(min + Math.random() * (max - min));
  return (
    <div className={styles.container}>
      <div className={styles.top}>
        <div className={styles.div1}></div>
        <div className={styles.div2}></div>
      </div>
      <div className={styles.mid}>
        <div className={styles.div3}>
          <p>ABOUT TODO</p>
        </div>
        <div className={styles.div4}>
          <p>
            Welcome to my Todo App! a place when you can add your todos and
            catch Pokemon!
          </p>
        </div>
        <div className={styles.div5}>
          <img
            src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokemonId}.png`}
            alt={`pokemon id: ${rand}`}
            height={250}
            width={250}
          ></img>
        </div>
      </div>
    </div>
  );
}
