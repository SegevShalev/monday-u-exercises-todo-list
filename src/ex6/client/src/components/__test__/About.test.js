import React, { useState } from "react";
import renderer from "react-test-renderer";

import About from "../About";

test("render the About Component", () => {
  const tree = renderer.create(<About />).toJSON();
  expect(tree).toMatchSnapshot();
});
