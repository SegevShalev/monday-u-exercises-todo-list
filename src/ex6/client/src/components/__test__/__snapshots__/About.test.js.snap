// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the About Component 1`] = `
<div
  className="container"
>
  <div
    className="top"
  >
    <div
      className="div1"
    />
    <div
      className="div2"
    />
  </div>
  <div
    className="mid"
  >
    <div
      className="div3"
    >
      <p>
        ABOUT TODO
      </p>
    </div>
    <div
      className="div4"
    >
      <p>
        Welcome to my Todo App! a place when you can add your todos and catch Pokemon!
      </p>
    </div>
    <div
      className="div5"
    >
      <img
        alt="pokemon id: undefined"
        height={250}
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
        width={250}
      />
    </div>
  </div>
</div>
`;
